### PR TITLE
Feature/replace deprecated container call

### PR DESCRIPTION
**Pull request by Chris53897**
Hi. Proof of concept for my (thats another account from me) idea in #65
A lot of code changes just to support symfony 4.4 :(
It works according to the test pipeline.
I don't have PHP 7.1 on my pc to verify it local.

If you find a simpler and better solution i would be glad.

Greetz

### CODE REVIEW

File: tests/BcLayer/BcLayerWebTestCase.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Mtarld\ApiPlatformMsBundle\Tests\BcLayer;
```
Comment (mtarld) [inline tests/BcLayer/BcLayerWebTestCase.php:3]: I'd rather have put those classes in `Mtarld\ApiPlatformMsBundle\Tests`

File: tests/BcLayer/BcLayerWebTestCase.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+// Remove after support of symfony support of 4.4 is dropped
+if (Kernel::VERSION_ID <= 50000) {
```
Comment (mtarld) [inline tests/BcLayer/BcLayerWebTestCase.php:12]: I don't like having conditional class definitions (and checking the kernel version id). WDYT about:
```php
<?php

namespace Mtarld\ApiPlatformMsBundle\Tests;

use LogicException;
use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
use Symfony\Component\DependencyInjection\ContainerInterface;
use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;

// Remove after support of symfony support of 4.4 is dropped
abstract class BcLayerWebTestCase extends WebTestCase
{
    private static $kernelContainer;

    protected static function bootKernel(array $options = [])
    {
        $kernel = parent::bootKernel($options);

        self::$kernelContainer = $kernel->getContainer();

        return $kernel;
    }

    protected static function getContainer(): ContainerInterface
    {
        if (method_exists(parent::class, 'getContainer')) {
            return parent::getContainer();
        }

        if (!static::$booted) {
            static::bootKernel();
        }

        try {
            return self::$kernelContainer->get('test.service_container');
        } catch (ServiceNotFoundException $e) {
            throw new LogicException('Could not find service "test.service_container". Try updating the "framework.test" config to "true".', 0, $e);
        }
    }
}
```

(Same for BcLayerKernelTestCase)

**Review by mtarld — CHANGES_REQUESTED** [review]

File: tests/BcLayer/BcLayerWebTestCase.php (hunk 2 above)
Comment (Chris53897) [reply tests/BcLayer/BcLayerWebTestCase.php]: Sorry for late response (newborn baby).
Thanks for looking into this.
I changed it to your idea.

**Comment by Chris53897** [comment]
Test for PHP 7.4 8.0 and 8.1 fail. info: First run after 5.4 is stable.
7.4 and 8.0 for Remaining self deprecation notices
8.1 because of api-platform

I guess these should be handled in an different PR. WDYT?

**Comment by mtarld** [comment]
Yes indeed. Let's do that!

**Comment by mtarld** [comment]
Here we are: https://github.com/mtarld/api-platform-ms-bundle/pull/68

**Review by mtarld — APPROVED** [review]
Thank you @Chris53897!

**Comment by Chris53897** [comment]
@mtarld Could you please aprove the workflow to run the github action tests. I guess i did the rebase wrong. Sorry for that.

**Comment by mtarld** [comment]
Thank you @Chris53897!

**Pull request merged**